### ISSUE TITLE
Add keyboard navigation support for Dashboard

### ISSUE THREAD

**Issue by nijikokun** [getinsomnia/insomnia#3183]
**Why**

Improved accessibility and navigation

**Expectations**

- Ability to use keyboard keys (up/down/left/right) arrows to navigate between cards
- Ability to use tab to switch between filter / create / dashboard cards
- Ability to use enter to enter dashboard card
- Ability to use another keybind to enter dashboard card action menu (three dots) 
- Auto-focus filter input when typing in dashboard

**Comment by ajtran303** [comment]
This is a related issue: https://github.com/Kong/insomnia/issues/1436

**Comment by ajtran303** [comment]
In addition to the keyboard navigation, there are already some documented bugs that make this app hard to use for developers using screen readers -- see more details in #1436 

Even as a sighted developer, I use a lot of keyboard navigation and there are some UI elements that don't obviously indicate when I am navigating over them. 

I think Insomnia could be a really good HTTP client that is more accessible than [the more popular ones](https://github.com/postmanlabs/postman-app-support/issues/3121) and useful for blind developers.

I bring this up because I think that even if a PR addresses this issue of keyboard navigation - there are still issues of elements not being labeled in such a way that even if someone is "on" an element with the keyboard, there's not an indication -- for both sighted and blind developers.

**Comment by nijikokun** [comment]
> I bring this up because I think that even if a PR addresses this issue of keyboard navigation - there are still issues of elements not being labeled in such a way that even if someone is "on" an element with the keyboard, there's not an indication -- for both sighted and blind developers.

We would love assistance in breaking these down and how we can best help in this regard. I definitely agree we could do a better job 🤗

**Comment by zersiax** [comment]
@nijikokun what kind of assistance do you require? Do you have specific questions?

**Comment by dimitropoulos** [comment]
> Do you have specific questions?

I don't think so at this time.

While it isn't everybody's cup of tea (after all, it's practically a whole different set of skills) I personally would love to see better accessibility in insomnia, and I have done this kind of work in the past.  In this case it's going to mean design work for focus states as well as a thorough overhaul on the UX side for determining what the tab order of things should be, and then followed by implementation.  cc: @erickrico
